src/ucm/util/replace.c

```diff
@@ -151,13 +159,28 @@ int ucm_orig_brk(void *addr)
     }
 }
 
+_UCM_DEFINE_DLSYM_FUNC(sbrk, ucm_orig_dlsym_sbrk, ucm_override_sbrk,
+                       void*, MAP_FAILED, intptr_t)
+
+void *ucm_orig_sbrk(intptr_t increment)
+{
+    if (!ucm_global_opts.enable_syscall) {
+        return ucm_orig_dlsym_sbrk(increment);
+    } else {
+        return ucm_orig_brk(ucm_orig_dlsym_sbrk(0) + increment) ?
```

Comment:
maybe we can call ucm_orig_dlsym_sbrk(0) only once?
or even use __curbrk , if exists?

Reply:
orig_sym_sbrk should be called to initialize __curbrk
optimized orig_sbrk to call dlsym only once